File: core/repository/import_graph_engine.py
from __future__ import annotations

from typing import Any, Dict
import re


def build_import_graph(text: str) -> Dict[str, Any]:
    src = text or ""
    py = re.findall(r"^\s*(?:import|from)\s+([A-Za-z0-9_.]+)", src, flags=re.MULTILINE)
    js = re.findall(r"""import\s+.*?from\s+['"]([^'"]+)['"]""", src)
    dart = re.findall(r"""import\s+['"]([^'"]+)['"]""", src)
    java = re.findall(r"^\s*import\s+([A-Za-z0-9_.]+);", src, flags=re.MULTILINE)
    nodes = sorted(set(py + js + dart + java))
    edges = [{"from": nodes[i], "to": nodes[i + 1]} for i in range(max(0, len(nodes) - 1))]
    return {"nodes": nodes, "edges": edges}

File: src/repository/importGraphEngine.ts
/**
 * Converted from Python: core/repository/import_graph_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function buildImportGraph(text: any): any {
  var src: any = py.or2(text, () => (""));
  var py_: any = py.reFindall("^\\s*(?:import|from)\\s+([A-Za-z0-9_.]+)", src, "m");
  var js: any = py.reFindall("import\\s+.*?from\\s+['\"]([^'\"]+)['\"]", src, "");
  var dart: any = py.reFindall("import\\s+['\"]([^'\"]+)['\"]", src, "");
  var java: any = py.reFindall("^\\s*import\\s+([A-Za-z0-9_.]+);", src, "m");
  var nodes: any = py.sorted(py.toSet(py.add(py.add(py.add(py_, js), dart), java)));
  var edges: any = py.range(py.max([0, py.sub(py.len(nodes), 1)])).map((i: any) => ({"from": py.at(nodes, i), "to": py.at(nodes, py.add(i, 1))}));
  return {"nodes": nodes, "edges": edges};
}
